Round Trip Integration Tests Don't Check Array Contents.

### Steps to Reproduce
Note: This is all under `pynwb\tests\integration\ui_write\base.py` where pynwb is wherever the repository lives.
The environment was created following the tutorial for running tests in pynwb (https://pynwb.readthedocs.io/en/stable/getting_started.html#installation)

[Relevant MatNWB Issue](https://github.com/NeurodataWithoutBorders/matnwb/issues/106#issuecomment-461572136)
1) `TestMapNWBContainer.assertContainerEqual()` expects the first argument to be the _actual_ read data and the second argument to be the _expected_ data.  This is flipped in the subclass `TestMapRoundTrip` when `assertContainerEqual` is called at the end of `test_roundtrip()`.
2) Fixing the above inconsistency revealed that h5py.Dataset classes are not accounted for when validation occurs.  The error message snippet looks like this:
```
FAIL: test_roundtrip (ui_write.test_base.TestTimeSeriesIO) (container_type='TimeSeries', nwbfield='data')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "...\pynwb\tests\integration\ui_write\base.py", line 129, in assertContainerEqual
    self.assertEqual(f1, f2)
AssertionError: <HDF5 dataset "data": shape (10,), type "<i4"> != [100, 110, 120, 130, 140, 150, 160, 170, 180, 190]
```
The reason why this didn't break tests earlier was because if the first argument is array data, then it will only validate if its contents are NWBContainers or NWBData.  In the expected data, the contents were numeric.
### Environment
Please describe your environment according to the following bullet points.
- **Python Executable:** Miniconda
- **Python Version:** Python 3.7.1
- **Operating System:** Windows 10
- **Pynwb Version:** origin/dev f21aa5d8

--- a/tests/integration/ui_write/base.py
+++ b/tests/integration/ui_write/base.py
@@ -3,6 +3,7 @@ from datetime import datetime
 from dateutil.tz import tzlocal, tzutc
 import os
 import numpy as np
+import h5py
 import numpy.testing as npt
 
 from pynwb import NWBContainer, get_manager, NWBFile, NWBData
@@ -123,7 +124,7 @@ class TestMapNWBContainer(unittest.TestCase):
                     elif isinstance(f2, NWBData):
                         self.assertTrue(np.array_equal(f1.data, f2))
                 else:
-                    if isinstance(f1, (float, np.float32, np.float16)):
+                    if isinstance(f1, (float, np.float32, np.float16, h5py.Dataset)):
                         npt.assert_almost_equal(f1, f2)
                     else:
                         self.assertEqual(f1, f2)
@@ -181,7 +182,7 @@ class TestMapRoundTrip(TestMapNWBContainer):
         # make sure we get a completely new object
         str(self.container)  # added as a test to make sure printing works
         self.assertNotEqual(id(self.container), id(self.read_container))
-        self.assertContainerEqual(self.container, self.read_container)
+        self.assertContainerEqual(self.read_container, self.container)
 
     def addContainer(self, nwbfile):
         ''' Should take an NWBFile object and add the container to it '''

